EssentialsProtect/src/com/earth2me/essentials/protect/EssentialsProtectBlockListener.java

```diff
@@ -90,4 +92,44 @@ public void onPortalLight(PortalCreateEvent event) {
             event.setCancelled(prot.getSettingBool(ProtectConfig.prevent_portal_creation));
         }
     }
+
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onSignDye(PlayerInteractEvent event) {
+        Material clickedBlock = event.getClickedBlock().getBlockData().getMaterial();
+        Action action = event.getAction();
+        Material itemInHandMat = event.getPlayer().getInventory().getItemInMainHand().getType();
+
+        // Check Right Click Action with Dye on Sign
+        if ((clickedBlock == Material.ACACIA_SIGN ||
+                clickedBlock == Material.ACACIA_WALL_SIGN ||
+                clickedBlock == Material.SPRUCE_SIGN ||
+                clickedBlock == Material.SPRUCE_WALL_SIGN ||
+                clickedBlock == Material.BIRCH_SIGN ||
+                clickedBlock == Material.BIRCH_WALL_SIGN ||
+                clickedBlock == Material.DARK_OAK_SIGN ||
+                clickedBlock == Material.DARK_OAK_WALL_SIGN ||
+                clickedBlock == Material.JUNGLE_SIGN ||
+                clickedBlock == Material.JUNGLE_WALL_SIGN ||
+                clickedBlock == Material.OAK_SIGN ||
+                clickedBlock == Material.OAK_WALL_SIGN) &&
+                action == Action.RIGHT_CLICK_BLOCK &&
+                (itemInHandMat == Material.BLACK_DYE ||
+                        itemInHandMat == Material.BLUE_DYE ||
+                        itemInHandMat == Material.BROWN_DYE ||
+                        itemInHandMat == Material.CYAN_DYE ||
+                        itemInHandMat == Material.GRAY_DYE ||
+                        itemInHandMat == Material.GREEN_DYE ||
+                        itemInHandMat == Material.LIGHT_BLUE_DYE ||
+                        itemInHandMat == Material.LIGHT_GRAY_DYE ||
+                        itemInHandMat == Material.LIME_DYE ||
+                        itemInHandMat == Material.MAGENTA_DYE ||
+                        itemInHandMat == Material.ORANGE_DYE ||
+                        itemInHandMat == Material.PINK_DYE ||
+                        itemInHandMat == Material.PURPLE_DYE ||
+                        itemInHandMat == Material.RED_DYE ||
+                        itemInHandMat == Material.WHITE_DYE ||
+                        itemInHandMat == Material.YELLOW_DYE)) {

```

Comment:
You should use the [MaterialUtil](https://github.com/EssentialsX/Essentials/blob/2.x/Essentials/src/com/earth2me/essentials/utils/MaterialUtil.java) class, which you can check easily items.